src/git/repo.rs

```diff
@@ -965,6 +984,24 @@ Either create it, or update the main branch setting by running:
             .map_err(wrap_git_error)?;
         Ok(make_non_zero_oid(oid))
     }
+
+    /// Returns whether the repo currently has changes staged to the index.
+    pub fn has_staged_changes(&self) -> eyre::Result<bool> {
+        let mut opts = StatusOptions::new();
+        opts.include_untracked(true);
```

Comment:
I don't think we should include untracked files for the amend operation in general, as they may be very large or contain sensitive information. And in any event, untracked files can't contribute to the staged changes, can they?